Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -4144,21 +4234,29 @@ bool IWindowExplorer.AddSwitchWindowAction(string Title)
 
         void IWindowExplorer.HighLightElement(ElementInfo ElementInfo)
         {
-            unhighlightLast();
+            UnhighlightLast();
 
-            if (string.IsNullOrEmpty(ElementInfo.XPath))
-            {
-                Driver.SwitchTo().DefaultContent();
-                SwitchFrame(ElementInfo.Path, ElementInfo.XPath, true);
-                ElementInfo.XPath = GenerateXpathForIWebElement((IWebElement)ElementInfo.ElementObject, "");
-            }
-            else
+            Driver.SwitchTo().DefaultContent();
+            SwitchFrame(ElementInfo.Path, ElementInfo.XPath, true);
+
+            ElementLocator elementXpathLocator = ElementInfo.Locators.Where(x => x.LocateBy == eLocateBy.ByXPath).FirstOrDefault();
+            if (elementXpathLocator == null || string.IsNullOrEmpty(elementXpathLocator.LocateValue))
             {
-                Driver.SwitchTo().DefaultContent();
-                SwitchFrame(ElementInfo.Path, ElementInfo.XPath, true);
+                if (string.IsNullOrEmpty(ElementInfo.XPath))
+                {
+                    ElementInfo.XPath = GenerateXpathForIWebElement((IWebElement)ElementInfo.ElementObject, "");
+                }
+                elementXpathLocator = new ElementLocator() { LocateBy = eLocateBy.ByXPath,LocateValue = ElementInfo.XPath};
+                // Add Generate function
+                ElementInfo.Locators.Add(elementXpathLocator);
             }
 
```

Comment:
why we must do all of this Xpath calculations even if we want to go by Element.Locators which might be don't enclude Xpath? can we run the method in 2 modes, use Xpath or Locators?